src/Jobs/UpdateAllProducts.php

```diff
@@ -36,7 +36,26 @@ public function get_name(): string {
 	 */
 	protected function process_items( array $items ) {
 		$products = $this->product_repository->find_by_ids( $items );
-
 		$this->product_syncer->update( $products );
 	}
+
+	/**
+	 * Schedules a delayed batched job
+	 *
+	 * @param int $delay The delay time in seconds
+	 */
+	public function schedule_delayed( int $delay ) {
+		if ( $this->can_schedule( [ 1 ] ) ) {
+			$this->action_scheduler->schedule_single( gmdate( 'U' ) + $delay, $this->get_create_batch_hook(), [ 1 ] );
+		}
+	}
+
+	/**
+	 * Check if the current job is currently running
+	 *
+	 * @return bool True if the job is running.
+	 */
+	public function is_syncing(): bool {
+		return $this->is_running( [ 1 ] );
```

Comment:
> Having looked at that class I think we could improve the search by adding the parameter 'group' => 'gla' otherwise we could potentially be looping through 1000's of pending subscriptions.

What do you mean with this? And where shall I add that group?

Reply:
As far as I see it's automatically added in the search method.

Reply:
Yup, you are right, I missed [the line where the group is added](https://github.com/woocommerce/google-listings-and-ads/blob/2.2.0/src/ActionScheduler/ActionScheduler.php#L147) in the ActionScheduler class.